### PR TITLE
[feat]: [PL-25416]: Git to Harness fix: Fix the issue where we are not able t…

### PR DESCRIPTION
**Pull request by DeepakPatankar**
…o get the commit of a file

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/33579)

### CODE REVIEW

**Review by bhavya181 — COMMENTED** [review]
> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33579)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by rathodmeetsatish — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33579)* status: 1 file(s) to review. 1 of 2 approvals obtained.

File: product/ci/scm/file/file.go [hunk 1]
```diff
@@ -57,6 +57,12 @@ func FindFile(ctx context.Context, fileRequest *pb.GetFileRequest, log *zap.Suga
 		content.Sha, "blob id", content.BlobID, "elapsed_time_ms", utils.TimeSince(start))
 
 	commitID := string(content.Sha)
+	// If the caller has provided a ref than we return that ref as the commitId
+	if fileRequest.GetRef() != "" {
```
Comment (meenakshi-harness) [inline product/ci/scm/file/file.go:61]: if this `fileRequest.GetRef()` is null ?

**Review by mohitg0795 — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33579#-:-N2za2wE-vIqPetnkOH9:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**